source/common/config/subscription_factory.h

```diff
@@ -37,19 +37,18 @@ class SubscriptionFactory {
    *        service description).
    * @param api reference to the Api object
    */
-  template <class ResourceType>
-  static std::unique_ptr<Subscription<ResourceType>> subscriptionFromConfigSource(
+  static std::unique_ptr<Subscription> subscriptionFromConfigSource(
       const envoy::api::v2::core::ConfigSource& config, const LocalInfo::LocalInfo& local_info,
       Event::Dispatcher& dispatcher, Upstream::ClusterManager& cm, Runtime::RandomGenerator& random,
       Stats::Scope& scope, const std::string& rest_method, const std::string& grpc_method,
-      Api::Api& api) {
-    std::unique_ptr<Subscription<ResourceType>> result;
+      const std::string& type_url, Api::Api& api) {
```

Comment:
Nit: why not `string_view`? You seem to be making that improvement elsewhere :)

Reply:
Thanks, good catch.